### ISSUE TITLE
Black Screen on kernel 4.7.5-200

### ISSUE THREAD

**Issue by EpicJhon** [imageguy/fglrx-for-Fedora#7]
I tried to install, but does not work in the new kernel 4.7.5-200

**Comment by EpicJhon** [comment]
that point does not pass

![image](https://cloud.githubusercontent.com/assets/2496458/19023288/18d84980-88b1-11e6-9bd9-9c4115b141ad.png)

**Comment by mirh** [comment]
Still an issue?

**Comment by bykbtzr** [comment]
@mirh I think so, if I remember correctly, I ran into the same issue about a month ago

**Comment by imageguy** [comment]
Yes, still an issue. It's on my "list of things to look at", but at this point I just live with it, though it's annoying.  The login screen is there in a sense, it just doesn't show. If you press enter, it will log you into whatever account was used last. Nothing will show until you type in the password, after that my KDE starst up.

**Comment by mirh** [comment]
Wait a minute
Described this way, I have the exact fucking same in Manjaro. With xfce. 

_Sometimes annoying_

**Comment by imageguy** [comment]
If I understand it properly, the login screen is independent of the desktop you run, be it KDE, Gnome, Xfce. I've saved relevant logs but haven't had a chance to really work on it. A long while ago, I tried the process that was outlined in one of the entries on the community discussion site, but it didn't work for me. Most likely I messed something up, but hadn't had time to really look at it then.

**Comment by imageguy** [comment]
....though I'd say _often_ annoying instead of _sometimes_ annoying.....

**Comment by mirh** [comment]
> If I understand it properly, the login screen is independent of the desktop you run, be it KDE, Gnome, Xfce.

Mhh, you have a point, it's handled by the display manager. 
And quite funnily intriguinly both KDE and xfce use SDDM. 

Threfore this situation can only make sense. 
It would be interesting to compare with GDM and LightDM then.

**Comment by mirh** [comment]
***Ohhh***, it starts to get always more and more interesting. 
https://aur.archlinux.org/cgit/aur.git/tree/arch-fglrx-authatieventsd_new.patch?h=catalyst-utils
Notice how there are GDM and LDM entries for, well, _fixing something_. 
Which eventually seems to derive from [these](https://bugs.launchpad.net/ubuntu/+source/fglrx-installer/+bug/38915/comments/87) [problems](https://bugs.launchpad.net/ubuntu/+source/linux-restricted-modules-2.6.20/+bug/112518/comments/33). Very similar to what we have. 
Also [this](https://aur.archlinux.org/packages/catalyst-fix-gdm) is worth investigating. 

Last but not least then, are you perhaps getting `[fglrx:firegl_unlock] *ERROR* firegl_lock_free failed!` too?

**Comment by imageguy** [comment]
According to my /var/log/messages, mine fails in shader compilation:
--------------------
Sep 12 17:20:40 localhost journal[1636]: Shader compilation failed:#012Fragment shader failed to compile with the following errors:#012ERROR: error(#271) Explicit version number 120 not supported by GL3 forward compatible context#012ERROR: error(#273) 1 compilation errors.  No code generated#012
Sep 12 17:20:40 localhost journal[1636]: Failed to link GLSL program:#012Vertex and Fragment shader(s) were not successfully compiled before glLinkProgram() was called.  Link failed. #012
Sep 12 17:20:40 localhost journal[1636]: driver/gl/cogl-pipeline-progend-glsl.c:384: GL error (1282): Invalid operation
-------------------------
There are a couple of patches for cogl that are supposed to fix this problem (see https://bugzilla.gnome.org/show_bug.cgi?id=756306). The first time i tried, long ago, I messed something up and cogl just wouldn't work any more. I just tried again, cogl works, but no change in behavior. Based on the comments at the bottom of the page,  patches might not be good or enough, but I haven't had a chance to study the thread yet.

**Comment by imageguy** [comment]
I haven't seen the fglrx_lock_free error, at least not in messages.

**Comment by mirh** [comment]
Oh, that's in dmesg. 
~~I had never checked /var/log/messages then~~  *cause that's all in journalctl in arch*

---
Give a check to 
https://bugzilla.gnome.org/show_bug.cgi?id=771587
https://bbs.archlinux.org/viewtopic.php?id=57084&p=287#p1571522

**Comment by imageguy** [comment]
As far as I understand, dmesg stuff, at least on Fedora, is one of the things going into /var/log/messages. I checked and there are no fglrx_lock_free messages on my box.

The version issue is apparently at the root of the problem - it's supposed to be addressed in  gl_Add_a_quirk_for_the_GLSL_version_to_use_on_AMD_v2.patch  in https://bugzilla.gnome.org/show_bug.cgi?id=756306, but that code seems slightly different than the one in 771587 and has an earlier date.

I took a peek today and I think my cogl patch didn't install, or probably installed in a wrong spot, when I did a make install. The README refers to the "INSTALL" document, which is nowhere to be found. I'll take a peek at the Makefile and if all fails try copying stuff over by hand and see what gives.

**Comment by imageguy** [comment]
No luck luck with the patches. I applied both patches from 756306, then added the patch from 771587 on top - it did add an extra piece to the other patches. I then rebuilt my fglrx, but no luck, I still get the version failure. I did make sure it's installed in /usr/lib64 (/lib64 links to it on my box) so, as far as I know, I am picking up the patches. I'll poke some more as I have time and if I get any ideas...

**Comment by mirh** [comment]
~~Mhh, I dunno, maybe time has brought in some new breakage?
Can you downgrade easily on fedora?~~

Mhh, Vi0L0 was indeed reporting back that KDE behaves differently. 
In other news, I just double checked journal and xorg.log and I'm not getting that error. Which possibly implies it has nothing to do with black screen. 
Maybe we should open another issue for that?

**Comment by imageguy** [comment]
I am not 100% sure that my "fix" was good, though I do know this error is one of the causes of the black login screen, and it is at the right time in the log. I have a feeling there are several possible causes of the black login screen - it looks like yours is different than mine. You might try opening an issue for that....
